tests/subsys/mgmt/zcbor_bulk/src/main.c

```diff
@@ -0,0 +1,224 @@
+
+	zassert_equal(rc, 0, "Expected 0, got %d", rc);
+	zassert_equal(decoded, ARRAY_SIZE(dm), "Expected %d got %d",
+		ARRAY_SIZE(dm), decoded);
+	zassert_equal(one, 1, "Expected 1");
+	zassert_equal(sizeof("world") - 1, world.len, "Expected length %d\n",
+		sizeof("world") - 1);
+	zassert_equal(0, memcmp(world.value, "world", world.len),
+		"Expected \"world\", got %.*s", world.len, world.value);
+	zassert_true(bool_val, "Expected bool_val == true");
+}
+
+void test_not_map(void)
+{
+	uint8_t buffer[512];
+	struct zcbor_string world;
+	uint32_t one = 0;
+	bool bool_val = false;
+	zcbor_state_t zsd[4] = { 0 };
+	size_t decoded = 1111;
+	bool ok;
+	struct zcbor_map_decode_key_val dm[] = {
+		ZCBOR_MAP_DECODE_KEY_VAL(hello, zcbor_tstr_decode, &world),
+		ZCBOR_MAP_DECODE_KEY_VAL(one, zcbor_uint32_decode, &one),
+		ZCBOR_MAP_DECODE_KEY_VAL(bool_val, zcbor_bool_decode, &bool_val)
+	};
+
+	zcbor_new_encode_state(zsd, 2, buffer, ARRAY_SIZE(buffer), 0);
+
+	/* List "hello", "world" */
+	ok = zcbor_list_start_encode(zsd, 10) &&
+	     zcbor_tstr_put_lit(zsd, "hello") && zcbor_tstr_put_lit(zsd, "world")	&&
+	     zcbor_list_end_encode(zsd, 10);
+
+	zassert_true(ok, "Expected to be successful in encoding test pattern");
+
+	zassert_true(zcbor_new_decode_state(zsd, 4, buffer, ARRAY_SIZE(buffer), 1),
+		"Failed to init");
+
+	int rc = zcbor_map_decode_bulk(zsd, dm, ARRAY_SIZE(dm), &decoded);
+
+	zassert_equal(rc, -EBADMSG, "Expected -EBADMSG(%d), got %d", -EBADMSG, rc);
+	zassert_equal(decoded, 1111, "Expected unodified");
+}
+
+void test_bad_type(void)
+{
+	uint8_t buffer[512];
+	struct zcbor_string world = { 0 };
+	uint32_t one = 0;
+	bool bool_val = false;
+	zcbor_state_t zsd[4] = { 0 };
+	size_t decoded = 0;
+	bool ok;
+	struct zcbor_map_decode_key_val dm[] = {
+		/* First entry has bad decoder given instead of tstr */
+		ZCBOR_MAP_DECODE_KEY_VAL(hello, zcbor_uint32_decode, &world),
+		ZCBOR_MAP_DECODE_KEY_VAL(one, zcbor_uint32_decode, &one),
+		ZCBOR_MAP_DECODE_KEY_VAL(bool_val, zcbor_bool_decode, &bool_val)
+	};
+
+	zcbor_new_encode_state(zsd, 2, buffer, ARRAY_SIZE(buffer), 0);
+
+	/* { "hello":"world", "one":1, "bool_val":true } */
+	ok = zcbor_map_start_encode(zsd, 10) &&
+	     zcbor_tstr_put_lit(zsd, "hello") && zcbor_tstr_put_lit(zsd, "world")	&&
+	     zcbor_tstr_put_lit(zsd, "one") && zcbor_uint32_put(zsd, 1)			&&
+	     zcbor_tstr_put_lit(zsd, "bool_val") && zcbor_true_put(zsd)			&&
+	     zcbor_map_end_encode(zsd, 10);
+
+	zassert_true(ok, "Expected to be successful in encoding test pattern");
+
+	zassert_true(zcbor_new_decode_state(zsd, 4, buffer, ARRAY_SIZE(buffer), 1),
+		"Failed to init");
+
+	int rc = zcbor_map_decode_bulk(zsd, dm, ARRAY_SIZE(dm), &decoded);
+
+	zassert_equal(rc, -ENOMSG, "Expected -ENOMSG, got %d", rc);
+	zassert_equal(decoded, 0, "Expected %d got %d",
+		ARRAY_SIZE(dm), decoded);
+	zassert_equal(one, 0, "Expected 1");
+	zassert_equal(0, world.len, "Expected to be unmodified");
```

Comment:
expected string/world

Reply:
The world.len is expected to be untouched when decoding fails.